src/App.js

```diff
@@ -0,0 +1,143 @@
+        console.log("Error in getAllBoards()", err);
+      });
+  }, []);
+  
+  const handleNewBoardSubmit = (data) => {
+    backend
+    .createNewBoard(data)
+    .then((result) => {
+      setBoardData((prevBoardData) => [result, ...prevBoardData]);
+    })
+    .catch((err) => {
+      console.log("Error in handleNewBoardSubmit", err);
+    })
+  };
+  
+  const handleBoardSelect = (board) => {
+    setSelectedBoard(board);
+    backend
+    .getBoardCards(board.id)
+    .then((result) => {
+      setCardEntries(result);
+    })
+    .catch((err) => {
+      console.log("Error in handleBoardSelect", err)
+    })
+  };
+  
+  const handleBoardDelete = (boardId) => {
+    backend
+    .deleteBoard(boardId)
+    .then(setBoardData((prev) => prev.filter((board) => board.board_id !== boardId))
+    )
+    .catch((err) => {
+      console.log("Error in handleBoardDelete", err)
+    })
+  };
+  
+  const handleNewCardSubmit = (data) => {
+    backend
+    .createNewCard(data)
+    .then((result) => {
+      setCardEntries((prev) => [...prev, result])
+    })
+    .catch((err) => {
+      console.log("Error in handleNewCardSubmit", err)
+    })
+  };
+
+  const handleAddLike = (cardId) => {
+    backend
+    .addLike(cardId)
+    .then((result) => {
+      setCardEntries(cardEntries.map((card) => {
+        if (card.card_id === cardId) {
+          return {...card, likes_count:result.card_like_count};
+        } else {
+          return card;
+        };
+      }))
+    })
+    .catch((err) => {
+      console.log("Error in handleAddLike", err)
+    })
+  };
+
+  const handleDeleteCard = (cardId) => {
+    backend
+    .deleteCard(cardId)
+    .then(setCardEntries((prev) => prev.filter((card) => card.card_id !== cardId))
+    )
+    .catch((err) => {
+      console.log("Error in handleDelete Card", err)
+    });
+  };
+
+  return (
+    <div className="page__container">
+      <div className="content__container">
+        <h1>Inspiration Board</h1>
+        <section className="boards__container">
+          
+          <section className="BoardList__container">
+            <BoardList 
+              boardData={boardData} 
+              onBoardSelect={handleBoardSelect} 
+              handleBoardDelete={handleBoardDelete}
+
+            />
+          </section>
+
+          <section className="newBoardForm__container">
+            <NewBoardForm handleNewBoardSubmit={handleNewBoardSubmit} />
+          </section>
+        </section>
+        {selectedBoard ?
+          <section className="cards__container">
+            <section className="cardList__container">
+              <CardList
+                board = {selectedBoard}
+                cardEntries = {cardEntries}
+                onAddLike = {handleAddLike}
+                onDelete = {handleDeleteCard}
+              />
+            </section>
+
+            <section className="newCardForm__container">
+              <NewCardForm
+                boardId={selectedBoard.id}
```

Comment:
Rather than passing the selected board id down, only to have the NewCardForm pack it into the card data passed back up, consider omitting the `boardId` prop, and have the `handleNewCardSubmit` take care of adding in the selected board data itself.